published/external/xdpapi_helper.h

```diff
@@ -0,0 +1,289 @@
+//
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <xdpapi.h>
+
+#if defined(_KERNEL_MODE)
+
+static const XDP_API_CLIENT _XdpApiContext = {0};
+static const XDP_API_PROVIDER_DISPATCH *_XdpApi;
+static const XDP_API_PROVIDER_BINDING_CONTEXT *_ProviderBindingContext;
+
+#define API_ABSTRACTION(FunctionCall, ...)  \
```

Comment:
I think it's better to just write this code directly in each location

Reply:
Is it better to multiply define same functions for User and Kernel mode?